onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp

```diff
@@ -560,13 +576,97 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
+        }
+        // Create a zero point tensor, since it's a required input by DML
+        else
+        {
+            auto inputSizes = m_inputTensorDescs[0].GetSizes();
+            TensorDesc intermediateOutputTensorDesc = TensorDesc(
+                    GetMlDataTypeFromDmlDataType(inputDataType),
+                    inputSizes,
+                    inputSizes,
+                    TensorAxis::DoNotCoerce,
+                    TensorAxis::W,
+                    TensorAxis::RightAligned,
+                    NchwDimensionCount, // minDimensionCount
+                    0 // guaranteedBaseOffsetAlignment
+                );
+            DML_TENSOR_DESC namedIntermediateOutputTensorDesc = intermediateOutputTensorDesc.GetDmlDesc();
+
+            // Create a tensor full of zeros
+            DML_FILL_VALUE_CONSTANT_OPERATOR_DESC zerosDesc = {};
+            zerosDesc.ValueDataType = inputDataType;
+            zerosDesc.OutputTensor = &namedIntermediateOutputTensorDesc;
+            const DML_OPERATOR_DESC opDesc1 = { DML_OPERATOR_FILL_VALUE_CONSTANT, &zerosDesc };
+
+            TOperatorDesc qLinearDesc = {};
+            qLinearDesc.InputTensor = &inputDescs[OnnxInputIndex::inputIndex];
+            qLinearDesc.ScaleTensor = &inputDescs[OnnxInputIndex::scaleIndex];
+            qLinearDesc.ZeroPointTensor = zerosDesc.OutputTensor;
+            qLinearDesc.OutputTensor = &outputDescs[0];
+            const DML_OPERATOR_DESC opDesc2 = { ApiTraits::OperatorDescTraits<TOperatorDesc>::Type, &qLinearDesc};
+
+            // Construct the graph
+            DML_INPUT_GRAPH_EDGE_DESC inputEdges[2];
+            DML_INTERMEDIATE_GRAPH_EDGE_DESC intermediateEdges[1];
+            DML_OUTPUT_GRAPH_EDGE_DESC outputEdges[1];
+
+            MLOperatorGraphDesc operatorGraphDesc = {};
+            operatorGraphDesc.nodeCount = 2;
+            const DML_OPERATOR_DESC* opDescs[] = {&opDesc1, &opDesc2};
+            operatorGraphDesc.nodesAsOpDesc = std::data(opDescs);
+
+            const uint32_t fillValueNodeIndex = 0;
+            const uint32_t dequantizeNodeIndex = 1;
+
+            // Input edges
+            DML_INPUT_GRAPH_EDGE_DESC inputToDequantizeEdge = {};
+            inputToDequantizeEdge.GraphInputIndex = OnnxInputIndex::inputIndex;
+            inputToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex;
+            inputToDequantizeEdge.ToNodeInputIndex = 0;
+            inputEdges[0] = inputToDequantizeEdge;
+
+            DML_INPUT_GRAPH_EDGE_DESC scaleToDequantizeEdge = {};
+            scaleToDequantizeEdge.GraphInputIndex = OnnxInputIndex::scaleIndex;
+            scaleToDequantizeEdge.ToNodeIndex = dequantizeNodeIndex;
+            scaleToDequantizeEdge.ToNodeInputIndex = 1;
+            inputEdges[1] = scaleToDequantizeEdge;
+
+            operatorGraphDesc.inputEdgeCount = gsl::narrow_cast<uint32_t>(std::size(inputEdges));
+            operatorGraphDesc.inputEdges = std::data(inputEdges);
```

Comment:
If we're going to use `std::size` and `std::data`, we may as well use `std::array` instead. `std::size` and `std::data` are meant to be used with legacy code or code that you don't own, but there's no reason not to use `std::array` here.